cpp/src/arrow/compute/kernels/scalar_compare_test.cc

```diff
@@ -1850,5 +1851,154 @@ TEST(TestMaxElementWiseMinElementWise, CommonTemporal) {
+TYPED_TEST(TestNumericBetweenKernel, SimpleBetweenArrayScalarScalar) {
+  using ScalarType = typename TypeTraits<TypeParam>::ScalarType;
+  using CType = typename TypeTraits<TypeParam>::CType;
+
+  Datum zero(std::make_shared<ScalarType>(CType(0)));
+  Datum four(std::make_shared<ScalarType>(CType(4)));
+  ValidateBetween<TypeParam>("[]", zero, four, "[]");
+  ValidateBetween<TypeParam>("[null]", zero, four, "[null]");
+  ValidateBetween<TypeParam>("[0,0,1,1,2,2]", zero, four, "[0,0,1,1,1,1]");
+  ValidateBetween<TypeParam>("[0,1,2,3,4,5]", zero, four, "[0,1,1,1,0,0]");
+  ValidateBetween<TypeParam>("[5,4,3,2,1,0]", zero, four, "[0,0,1,1,1,0]");
+  ValidateBetween<TypeParam>("[null,0,1,1]", zero, four, "[null,0,1,1]");
+}
+
+TEST(TestSimpleBetweenKernel, SimpleStringTest) {
+  using ScalarType = typename TypeTraits<StringType>::ScalarType;
+  auto l = Datum(std::make_shared<ScalarType>("abc"));
+  auto r = Datum(std::make_shared<ScalarType>("zzz"));
+  ValidateBetween<StringType>("[]", l, r, "[]");
+  ValidateBetween<StringType>("[null]", l, r, "[null]");
+  ValidateBetween<StringType>(R"(["aaa", "aaaa", "ccc", "z"])", l, r,
+                              R"([false, false, true, true])");
+  ValidateBetween<StringType>(R"(["a", "aaaa", "c", "z"])", l, r,
+                              R"([false, false, true, true])");
+  ValidateBetween<StringType>(R"(["a", "aaaa", "fff", "zzzz"])", l, r,
+                              R"([false, false, true, false])");
+  ValidateBetween<StringType>(R"(["abd", null, null, "zzx"])", l, r,
+                              R"([true, null, null, true])");
+}
+
+TEST(TestSimpleBetweenKernel, SimpleTimestampTest) {
+  using ScalarType = typename TypeTraits<TimestampType>::ScalarType;
+  auto checkTimestampArray = [](std::shared_ptr<DataType> type, const char* input_str,
+                                const Datum& lhs, const Datum& rhs,
+                                const char* expected_str) {
+    auto value = ArrayFromJSON(type, input_str);
+    auto expected = ArrayFromJSON(boolean(), expected_str);
+    ValidateBetween<TimestampType>(value, lhs, rhs, expected);
+  };
+  auto unit = TimeUnit::SECOND;
+  auto l = Datum(std::make_shared<ScalarType>(923184000, timestamp(unit)));
+  auto r = Datum(std::make_shared<ScalarType>(1602032602, timestamp(unit)));
+  checkTimestampArray(timestamp(unit), "[]", l, r, "[]");
+  checkTimestampArray(timestamp(unit), "[null]", l, r, "[null]");
+  checkTimestampArray(timestamp(unit), R"(["1970-01-01","2000-02-29","1900-02-28"])", l,
+                      r, "[false,true,false]");
+  checkTimestampArray(timestamp(unit), R"(["1970-01-01","2000-02-29","2004-02-28"])", l,
+                      r, "[false,true,true]");
+  checkTimestampArray(timestamp(unit), R"(["2018-01-01","1999-04-04","1900-02-28"])", l,
+                      r, "[true,false,false]");
+}
+
+TYPED_TEST(TestNumericBetweenKernel, SimpleBetweenArrayArrayArray) {
```

Comment:
Could you move this to right after `TYPED_TEST(TestNumericBetweenKernel, SimpleBetweenArrayScalarScalar)`?

Reply:
Ok.